test/index.test.ts

```diff
@@ -96,3 +100,86 @@ test('source image cache pruning', async (t) => {
     'cache size should not be equal to number of images * image size if we exceed threshold'
   )
 })
+
+test('should add WAF headers to local images being transformed', async (t) => {
+  const handler = createIPXHandler({
+    basePath: '/_ipx/',
+    cacheDir: '/tmp/ipx-cache',
+    bypassDomainCheck: true
+  }, (sourceImageOptions) => {
+    t.assert(sourceImageOptions.requestHeaders && sourceImageOptions.requestHeaders['X-Nf-Waf-Bypass-Token'] === 'some token')
+
+    return Promise.resolve({ finalize: () => { } } as SourceImageResult)
+  })
+
+  await handler(
+    {
+      rawUrl: 'http://localhost:3000/some-path',
+      path: '/_ipx/w_500/no-file.jpg',
+      headers: { 'X-Nf-Waf-Bypass-Token': 'some token' },
+      rawQuery: '',
+      httpMethod: 'GET',
+      queryStringParameters: {},
+      multiValueQueryStringParameters: {},
+      multiValueHeaders: {},
+      isBase64Encoded: false,
+      body: null
+    },
+    getHandlerContext()
+  )
+})
+
+test('should not add WAF headers to local images being transformed', async (t) => {
```

Comment:
not sure if I read this right, but I think this should be:
```suggestion
test('should not add WAF headers to remote images being transformed', async (t) => {
```